src/widget/weffectchainpresetbutton.cpp

```diff
@@ -0,0 +1,122 @@
+#include "widget/weffectchainpresetbutton.h"
+
+#include <QCheckBox>
+#include <QWidgetAction>
+
+#include "effects/presets/effectpresetmanager.h"
+#include "widget/effectwidgetutils.h"
+
+WEffectChainPresetButton::WEffectChainPresetButton(QWidget* parent, EffectsManager* pEffectsManager)
+        : QPushButton(parent),
+          WBaseWidget(this),
+          m_pEffectsManager(pEffectsManager),
+          m_pChainPresetManager(pEffectsManager->getChainPresetManager()),
+          m_pMenu(make_parented<QMenu>(new QMenu(this))) {
+    setMenu(m_pMenu.get());
+    connect(m_pChainPresetManager.data(),
+            &EffectChainPresetManager::effectChainPresetListUpdated,
+            this,
+            &WEffectChainPresetButton::populateMenu);
+    setFocusPolicy(Qt::NoFocus);
+}
+
+void WEffectChainPresetButton::setup(const QDomNode& node, const SkinContext& context) {
+    m_iChainNumber = EffectWidgetUtils::getEffectUnitNumberFromNode(node, context);
+    m_pChain = EffectWidgetUtils::getEffectChainFromNode(
+            node, context, m_pEffectsManager);
+    for (const auto& pEffectSlot : m_pChain->getEffectSlots()) {
+        connect(pEffectSlot.data(),
+                &EffectSlot::effectChanged,
+                this,
+                &WEffectChainPresetButton::populateMenu);
+        connect(pEffectSlot.data(),
+                &EffectSlot::parametersChanged,
+                this,
+                &WEffectChainPresetButton::populateMenu);
+    }
+    populateMenu();
+    // TODO: enable applying external stylesheets
+}
+
+void WEffectChainPresetButton::populateMenu() {
+    m_pMenu->clear();
+
+    // Chain preset items
+    for (const auto& pChainPreset : m_pChainPresetManager->getPresetsSorted()) {
+        m_pMenu->addAction(pChainPreset->name(), this, [this, pChainPreset]() {
+            m_pChain->loadChainPreset(pChainPreset);
+        });
+    }
+    m_pMenu->addSeparator();
+    m_pMenu->addAction(tr("Save preset"), this, [this]() {
```

Comment:
```suggestion
    m_pMenu->addAction(tr("Save current state as ..."), this, [this]() {
```

Reply:
How about `Save as preset`?

Reply:
I changed this to "Save new preset": 642ebd1d5112bb135624733094e4098a98917793